src/LinkDotNet.Blog.Web/Controller/RssFeedController.cs

```diff
@@ -34,18 +35,28 @@ public RssFeedController(
 
         description = introductionConfiguration.Value.Description;
         blogName = applicationConfiguration.Value.BlogName;
+        blogPostsPerPage = applicationConfiguration.Value.BlogPostsPerPage;
         this.blogPostRepository = blogPostRepository;
     }
 
     [ResponseCache(Duration = 1200)]
     [HttpGet]
-    public async Task<IActionResult> GetRssFeed()
+    public async Task<IActionResult> GetRssFeed([FromQuery] bool withContent = false, [FromQuery] int? numberOfBlogPosts = null)
     {
+        if (!ModelState.IsValid)
+        {
+            return BadRequest(ModelState);
+        }
+
+        numberOfBlogPosts ??= blogPostsPerPage;
```

Comment:
I would move this bit into `GetBlogPostsItemsWithContent` as it is only used there.